### PR TITLE
 Multi-tenant docs update  issue-> #11187

### PR DESCRIPTION
**Pull request by khairnarsaurabh23**
Solved issue #11187 Improved useDb() docs

### CODE REVIEW

File: lib/connection.js [hunk 1]
```diff
@@ -1388,6 +1388,21 @@ Connection.prototype.syncIndexes = async function syncIndexes() {
  * Switches to a different database using the same connection pool.
  *
  * Returns a new connection object, with the new db.
+ * 
+ * ####Example
+ *    async function createMultiTenatConnection(tenatId, modelName, schema) {
```
Comment (vkarpov15) [inline lib/connection.js:1393]: Spelling, "Tenat" -> "Tenant"

File: lib/connection.js [hunk 2]
```diff
@@ -1388,6 +1388,21 @@ Connection.prototype.syncIndexes = async function syncIndexes() {
...
+  *     const conn = await mongoose.createConnection('mongodb://localhost:27017/test');
```
Comment (vkarpov15) [inline lib/connection.js:1394]: Why is this line here? `conn` isn't used at all here.

File: lib/connection.js [hunk 3]
```diff
@@ -1388,6 +1388,21 @@ Connection.prototype.syncIndexes = async function syncIndexes() {
...
+  *     const dbName = `db_${tenatId}`;
+  *     const dbOptions = {
+    *     //If true, create one connection object with multiple calls to useDb(). Default id false
+    *     userCache:true 
+    *     //If true, db will not listen to the events on original connection. Default is false
+    *     noListener:true
+  *     }
+  *     const db = mongoose.connection.useDb(dbName, dbOptions);
+  *     db.model(modelname, schema);
```
Comment (vkarpov15) [inline lib/connection.js:1403]: Need to be careful with `db.model()` and `useCache`. This line may throw an error if reusing an existing connection. Add a check:

```javascript
if (db.models[modelname] == null) {
  db.model(modelname, schema);
}
```

File: lib/connection.js [hunk 4]
```diff
@@ -1388,6 +1388,21 @@ Connection.prototype.syncIndexes = async function syncIndexes() {
  * Switches to a different database using the same connection pool.
  *
  * Returns a new connection object, with the new db.
+ * 
+ * ####Example
+ *    async function createMultiTenatConnection(tenatId, modelName, schema) {
+  *     const conn = await mongoose.createConnection('mongodb://localhost:27017/test');
+  *     const dbName = `db_${tenatId}`;
+  *     const dbOptions = {
+    *     //If true, create one connection object with multiple calls to useDb(). Default id false
```
Comment (vkarpov15) [inline lib/connection.js:1397]: "id" -> "is"

**Review by vkarpov15 — CHANGES_REQUESTED** [review]

File: lib/connection.js (hunk 2 above)
Comment (khairnarsaurabh23) [reply lib/connection.js:1394]: I thought making a connection is necessary for better understanding of reader.

File: lib/connection.js (hunk 1 above)
Comment (khairnarsaurabh23) [reply lib/connection.js:1393]: Apologize, I'll make that correction.

File: lib/connection.js (hunk 3 above)
Comment (khairnarsaurabh23) [reply lib/connection.js:1403]: I'll make necessary correction for that.

**Comment by khairnarsaurabh23** [comment]
@vkarpov15  can you please review the changes I made?

File: lib/connection.js [hunk 5]
```diff
@@ -1392,6 +1392,25 @@ Connection.prototype.syncIndexes = async function syncIndexes() {
  * Switches to a different database using the same connection pool.
  *
  * Returns a new connection object, with the new db.
+ * 
+ * ####Example
+ *    function createMultiTenatConnection(tenantId, modelName, schema) {
+  *     mongoose.connect('mongodb://localhost:27017/test');
```
Comment (vkarpov15) [inline lib/connection.js:1398]: Shouldn't put `mongoose.connect()` here, you don't want to call `mongoose.connect()` every time you create a multi tenant connection

File: lib/connection.js [hunk 6]
```diff
@@ -1392,6 +1392,25 @@ Connection.prototype.syncIndexes = async function syncIndexes() {
  * Switches to a different database using the same connection pool.
  *
  * Returns a new connection object, with the new db.
+ * 
+ * ####Example
+ *    function createMultiTenatConnection(tenantId, modelName, schema) {
```
Comment (vkarpov15) [inline lib/connection.js:1397]: Spelling 'Tenat' -> 'Tenant'

File: lib/connection.js [hunk 7]
```diff
@@ -1392,6 +1392,25 @@ Connection.prototype.syncIndexes = async function syncIndexes() {
...
+  *     const dbOptions = {
+    *     //If true, create one connection object with multiple
+    *     //calls to useDb(). Default is false
+    *     userCache:true 
+    *     //If true, db will not listen to the events on original
+    *     // connection. Default is false
+    *     noListener:true
+  *     }
+  *     const db = mongoose.connection.useDb(dbName, dbOptions);
+  *     if(db.models[modelName] == null){
```
Comment (vkarpov15) [inline lib/connection.js:1409]: Since this function creates a connection, why is it creating a model?

**Review by vkarpov15 — CHANGES_REQUESTED** [review]

**Comment by khairnarsaurabh23** [comment]
@vkarpov15  is there anything wrong? can you please correct me if so?

**Comment by vkarpov15** [comment]
@khairnarsaurabh23 I made several comments on Feb 13, see https://github.com/Automattic/mongoose/pull/11232/files/5528a432823a0d8776aa99f9b15f71fabe46b2c4#diff-f35455f30798a56117f979b6ad64a99cbc915318f08795935d68c80ab2223743
